buildSrc/src/main/java/gradlebuild/JniNokeePlugin.java

```diff
@@ -0,0 +1,204 @@
+package gradlebuild;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import dev.nokee.language.cpp.CppSourceSet;
+import dev.nokee.platform.jni.JavaNativeInterfaceLibrary;
+import dev.nokee.runtime.nativebase.TargetMachine;
+import dev.nokee.runtime.nativebase.TargetMachineFactory;
+import groovy.util.Node;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.publish.PublishingExtension;
+import org.gradle.api.publish.maven.MavenPublication;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.language.cpp.tasks.CppCompile;
+import org.gradle.model.Mutate;
+import org.gradle.model.RuleSource;
+import org.gradle.nativeplatform.platform.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+import org.gradle.nativeplatform.tasks.LinkSharedLibrary;
+import org.gradle.nativeplatform.toolchain.Clang;
+import org.gradle.nativeplatform.toolchain.Gcc;
+import org.gradle.nativeplatform.toolchain.NativeToolChainRegistry;
+import org.gradle.nativeplatform.toolchain.VisualCpp;
+
+import java.io.File;
+import java.util.Set;
+
+@SuppressWarnings("UnstableApiUsage")
+public abstract class JniNokeePlugin implements Plugin<Project> {
```

Comment:
❓ Is there a reason the JNI tests are run as part of another `Test` task? Nokee wires everything for the main `Test` task. There wasn't any consideration to "remove" the wiring of the JNI into the main test task. We can create a new test task but all the wiring will have to be done manually. Nokee could provide a `jni-unit-test` plugin that allows users to declare any number of JNI test tasks but that is a bit outside of the current scope. The project could remove the extra test task in favour of the main test task however there may be a reason for the task to existing in the first place.

Reply:
We have some special tests where `-Xcheck:jni` output is treated as a reason for failure. These are the tests marked with `JniChecksEnabled`. For other tests that were introduced earlier, we did not enforce these checks yet, and they do fail.

One option to remove this distinction would be to make sure none of the current code emits any JNI warnings.